plugins/woocommerce-admin/client/products/sections/pricing-section.tsx

```diff
@@ -0,0 +1,143 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Link, useFormContext } from '@woocommerce/components';
+import { Product, SETTINGS_STORE_NAME } from '@woocommerce/data';
+import { recordEvent } from '@woocommerce/tracks';
+import { useContext } from '@wordpress/element';
+import { useSelect } from '@wordpress/data';
+import {
+	// @ts-expect-error `__experimentalInputControl` does exist.
+	__experimentalInputControl as InputControl,
+} from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import { ProductSectionLayout } from '../layout/product-section-layout';
+import { getInputControlProps } from './utils';
+import { ADMIN_URL } from '../../utils/admin-settings';
+import { CurrencyContext } from '../../lib/currency-context';
+import {
+	NUMBERS_AND_DECIMAL_SEPARATOR,
+	ONLY_ONE_DECIMAL_SEPARATOR,
+} from '../constants';
+
+export const PricingSection: React.FC = () => {
+	const { getInputProps, setValue } = useFormContext< Product >();
+	const { isResolving: isTaxSettingsResolving, taxSettings } = useSelect(
+		( select ) => {
+			const { getSettings, hasFinishedResolution } =
+				select( SETTINGS_STORE_NAME );
+			return {
+				isResolving: ! hasFinishedResolution( 'getSettings', [
+					'tax',
+				] ),
+				taxSettings: getSettings( 'tax' ).tax || {},
+			};
+		}
+	);
+	const pricesIncludeTax =
+		taxSettings.woocommerce_prices_include_tax === 'yes';
+	const context = useContext( CurrencyContext );
+	const { getCurrencyConfig } = context;
+	const { decimalSeparator } = getCurrencyConfig();
+	const priceValidation = ( value: string, name: string ) => {
+		// Build regex to strip out everything except digits, decimal point and minus sign.
+		const regex = new RegExp(
+			NUMBERS_AND_DECIMAL_SEPARATOR.replace( '%s', decimalSeparator ),
+			'g'
+		);
+		const decimalRegex = new RegExp(
+			ONLY_ONE_DECIMAL_SEPARATOR.replaceAll( '%s', decimalSeparator ),
+			'g'
+		);
+		const cleanValue = value
```

Comment:
If I change my decimal separator to `,` I'm no longer able to add numbers after the decimal.  E.g., `10,59` gets updated to `10,00`.

Reply:
This has been fixed in the commit 55c7c97